lib/client.ml

```diff
@@ -251,13 +251,19 @@ let service_accepted t = function
 
 let handle_auth_failure t _ = function
   | [] -> Error "no authentication method left"
-  | xs when List.mem "publickey" xs ->
-    let pub = Hostkey.pub_of_priv t.key in
-    let met = Ssh.Pubkey (pub, None) in
-    Ok ({ t with state = Userauth_request met },
-        [ Ssh.Msg_userauth_request (t.user, service, met) ],
-        [])
-  | _ -> Error "no supported authentication methods left"
+  | xs ->
+    if List.mem "publickey" xs then
+      let pub = Hostkey.pub_of_priv t.key in
+      match t.state with
```

Comment:
We can match on the 2nd argument instead. I made a PR to your branch as I'm not able to make a suggestion for some reason.